portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +34,68 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  private static final Gson gson = new Gson();
+  private ArrayList<String> comments;
+
+  @Override
+  public void init() {
+    comments = new ArrayList<>();
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello Jag!");
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Task> tasks = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("comment");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Task task = new Task(comment, timestamp);
```

Comment:
a Task is actually a comment. should we rename the class Comment?

Reply:
done.